### ISSUE TITLE
IndexOutOfBoundsException with RowFocusListener and empty Grid

### ISSUE THREAD

**Issue by yoshie82** [TatuLund/GridFastNavigation#56]
More a minor annoyance ^^
Issue only occurs, if you have an empty grid with the initialized extension and an added RowFocusListener.
If you change the selectionMode of the grid, an IndexOutOfBoundsException is thrown, when component is rendered.

Throws exception
````java
final Grid<String> grid = new Grid<>();
grid.addColumn(String::length).setCaption("length");

final FastNavigation<String> fastNavigation = new FastNavigation<>(grid);
grid.setSelectionMode(Grid.SelectionMode.MULTI); // after constructor

fastNavigation.addRowFocusListener(event -> log.info("{}", event.getRow()));
grid.setItems(Collections.emptyList());
//        grid.setItems(Collections.singleton("say hello"));
````

Works fine
````java
final Grid<String> grid = new Grid<>();
grid.addColumn(String::length).setCaption("length");

grid.setSelectionMode(Grid.SelectionMode.MULTI); // before constructor
final FastNavigation<String> fastNavigation = new FastNavigation<>(grid);

fastNavigation.addRowFocusListener(event -> log.info("{}", event.getRow()));
grid.setItems(Collections.emptyList());
//        grid.setItems(Collections.singleton("say hello"));
````

This all happens at client side when FocusTracker#notifyFocusMoved is called. I don't know the exactly reason, but currentRow is set to 0 in that case. 

So the serverRPC#getItemAt(0) in FastNavigation tries to load the first item of an empty collection.

````java
        registerRpc(new FastNavigationServerRPC() {

        	private T getItemAt(int rowIndex) { // << is 0 in that case
        		T myBean = null;
        		if (rowIndex >= 0) {
        			myBean = g.getDataCommunicator().fetchItemsWithRange(rowIndex, 1).get(0);
        		}
        		return myBean;
        	}
````

**Comment by TatuLund** [comment]
Fixed in versions 2.1.14 & 1.1.9

**Comment by TatuLund** [comment]
I would appreciate your rating & feedback of the add-on in the Directory, https://vaadin.com/directory/component/gridfastnavigation-add-on
